### ISSUE TITLE
About MuSGD

### ISSUE THREAD

**Issue by JoJoliking** [ultralytics/ultralytics#22235]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

I am very interested in the MuSGD Optimizer and how to coding it due to the usage of Moun is not ideal in my detection task training !

**Comment by UltralyticsAssistant** [comment]
👋 Hello @JoJoliking, thank you for your interest in Ultralytics and for the question about MuSGD 🚀 This is an automated response to help you get started quickly—an Ultralytics engineer will also assist you here soon.

We recommend a visit to the [Docs](https://docs.ultralytics.com/) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples and where many of the most common questions may already be answered.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug it.

If this is a custom training ❓ Question (e.g., optimizer choices like MuSGD vs. momentum variants), please provide as much information as possible to help us guide you:
-Exact Ultralytics version and environment details
-Your training command or Python code snippet
-Model and optimizer settings used
-Training logs and screenshots of metrics/curves
-Representative samples of your dataset
And please verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/) ✅

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.

## Upgrade

Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a **Python>=3.8** environment with **PyTorch>=1.8** to verify your issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

Once you share the requested details, we’ll help you evaluate optimizer options for your detection task and next steps 🔧📈

**Comment by Y-T-G** [comment]
The code for MuSGD should be released with YOLO26 models next month

**Comment by atztao** [comment]
> The code for MuSGD should be released with YOLO26 models next monthMuSGD 的代码将于下个月与 YOLO26 型号一起发布

Now, It’s already the "next month".

**Comment by glenn-jocher** [comment]
Thanks for the nudge—MuSGD will be released together with YOLO26; it isn’t public yet, so please track updates on the YOLO26 page. ([docs.ultralytics.com](https://docs.ultralytics.com/models/yolo26/))

If you want to prototype now, you can plug in your own optimizer by overriding `build_optimizer()` in a custom trainer, e.g.:

```python
from ultralytics import YOLO
from ultralytics.models.yolo.detect import DetectionTrainer
import torch

class MyTrainer(DetectionTrainer):
    def build_optimizer(self, model):
        return torch.optim.SGD(model.parameters(),
                               lr=self.args.lr0,
                               momentum=self.args.momentum,
                               weight_decay=self.args.weight_decay,
                               nesterov=True)

model = YOLO("yolo11n.pt")
model.train(trainer=MyTrainer, data="coco8.yaml", epochs=5, lr0=0.01, momentum=0.937, weight_decay=0.0005)
```

See Advanced Customization (Trainer) for integrating custom trainers and the `build_optimizer` reference for method details. ([docs.ultralytics.com](https://docs.ultralytics.com/usage/engine/))
